docs: IOMesh offline installation (#134)
iomesh/docs: docs/deploy/install-iomesh.md

    operator-85877979-xqtml                               1/1     Running   0          2m8s
    ```

IOMesh Cluster has been installed successfully. Please go to [Setup IOMesh](setup-iomesh.md).

[1]: http://iomesh.com/charts
[2]: http://www.iomesh.com/docs/installation/setup-iomesh-storage#setup-data-network

## Install IOMesh Offline

### Prepare Offline Installation Package
1. Download [IOMesh offline installation package](https://download.smartx.com/iomesh-offline-v0.11.0.tgz).

2. Decompress the offline installation package
    ```shell
    tar -xf  iomesh-offline-v0.11.0.tgz && cd iomesh-offline
    ```

### Load Images

> Load IOMesh images on each Kubernetes worker node, then choosing corresponding image load command according to your container runtime and container management tools.

Reviewer: ```suggestion
Load IOMesh images on each Kubernetes worker node, then execute the corresponding image load script according to your container runtime and container management tools. 
```

<!--DOCUSAURUS_CODE_TABS-->

<!--Docker-->
```shell
docker load --input ./images/iomesh-offline-images.tar
```
<!--Containerd-->
```shell
ctr --namespace k8s.io image import ./images/iomesh-offline-images.tar
```

<!--Podman-->
```shell
podman load --input ./images/iomesh-offline-images.tar
```

<!--END_DOCUSAURUS_CODE_TABS-->

### Install IOMesh Using Offline Installation Package

#### Install CSI Snapshotter

The [CSI snapshotter](https://github.com/kubernetes-csi/external-snapshotter) is part of Kubernetes implementation of Container Storage Interface (CSI).
Instasll CSI snapshotter to enable the Volume Snapshot feature.

> **_NOTE_: CSI Snapshotter should be installed once per cluster**

> 1. Install Snapshot CRDs:

Reviewer: ```suggestion
1. Install Snapshot CRDs.
```

    ```shell
    kubectl create -f ./charts/external-snapshotter/config/crd
    ```

> 2. Install Snapshot Controller:

Reviewer: ```suggestion
2. Install Snapshot Controller.
```

    ```shell
    kubectl apply -n kube-system -f ./charts/external-snapshotter/deploy/kubernetes/snapshot-controller
    ```

> 3. Wait until the Snapshot Controller is ready:

Reviewer: ```suggestion
3. Wait until Snapshot Controller is ready.
```

    ```shell
    kubectl get sts snapshot-controller -n kube-system
    ```

    ```output
    NAME                  READY   AGE
    snapshot-controller   1/1     32s
   ```

#### Install IOMesh

> 1. Download `iomesh.yaml` with default configurations:

Reviewer: ```suggestion
1. Download `iomesh.yaml` with default configurations.
```

    ```shell
    ./helm show values charts/iomesh > iomesh.yaml
    ```

2. Customize `iomesh.yaml`.

   **(required)** Fill in `dataCIDR` according to your network:

    ```yaml
    iomesh:
      chunk:
        dataCIDR: "10.234.1.0/24" # change to your own data network CIDR
    ```

   **(required)** For Kubernetes worker node OS is `CentOS8` or `CoreOS`, set `mountIscsiLock` to `true`. Otherwise, set it to `false`:

    ```yaml
    csi-driver:
      driver:
        node:
          driver:
            mountIscsiLock: true
    ```

>    **(optional)** If you only want IOMesh to use a part of k8s node's disks, 

Reviewer: ```suggestion
 **(optional)** If you simply want IOMesh to utilize part of Kubernetes nodes, configure the labels of nodes to be utilized in `chunk.podPolicy.affinity`. For example:
```

   configure the specific node's label in the `chunk.podPolicy.affinity`, for

>    example:

Reviewer: ```suggestion
    **(optional)** If you simply want IOMesh to utilize part of Kubernetes nodes, configure the labels of nodes to be utilized in `chunk.podPolicy.affinity`. For example:
```

   
   ```yaml
   iomesh:
     chunk:
       podPolicy:
         affinity:
           nodeAffinity:
             requiredDuringSchedulingIgnoredDuringExecution:
               nodeSelectorTerms:
               - matchExpressions:
                 - key: kubernetes.io/hostname # specific node label's key
                   operator: In
                   values:
                   - iomesh-worker-0 # specific node label's value
                   - iomesh-worker-1
    ```
    For more information about pod affinity configuration rule, see: [pod affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity)

> 3. Install IOMesh Cluster:

Reviewer: ```suggestion
3. Install the IOMesh cluster.
```

    ```shell
    ./helm install iomesh ./charts/iomesh \
        --create-namespace \
        --namespace iomesh-system \
        --values iomesh.yaml \
        --wait
    ```

    ```output
    NAME: iomesh
    LAST DEPLOYED: Wed Jun 30 16:00:32 2021
    NAMESPACE: iomesh-system
    STATUS: deployed
    REVISION: 1
    TEST SUITE: None
    ```

> 4. You can run `kubectl --namespace iomesh-system get pods` to check the result:

Reviewer: ```suggestion
4. Run `kubectl --namespace iomesh-system get pods` to check the result.
```

    operator-6b87858cbd-sf72b                                 1/1     Running   0          12m
    ```

IOMesh Cluster has been installed successfully. Please go to [Setup IOMesh](setup-iomesh.md).

[1]: http://iomesh.com/charts
[2]: http://www.iomesh.com/docs/installation/setup-iomesh-storage#setup-data-network

## Install IOMesh Offline

### Prepare Offline Installation Package
1. Download [IOMesh offline installation package](https://download.smartx.com/iomesh-offline-v0.11.0.tgz).

2. Decompress the offline installation package
    ```shell

>     tar -xf  iomesh-offline-v0.11.0.tgz && cd iomesh-offline

Reviewer 2: 目录名需要带上版本号

Author: 确实安装包和目录都带上版本号更好一点，不过目前安装包已发布，要不下个版本的安装包再修正这个细节？
